modules/diskq/logqueue-disk-non-reliable.c

```diff
@@ -312,14 +319,18 @@ _pop_head(LogQueue *s, LogPathOptions *path_options)
     }
 
 success:
-  _maybe_move_messages_among_queue_segments(self);
+  if (!_maybe_move_messages_among_queue_segments(self))
+    {
+      stats_update = FALSE;
+    }
 
   g_mutex_unlock(&s->lock);
 
   if (s->use_backlog)
     _push_tail_qbacklog(self, msg, path_options);
 
-  log_queue_queued_messages_dec(s);
+  if (stats_update)
+    log_queue_queued_messages_dec(s);
```

Comment:
Of course, then this would be done under locking, which wanted to avoid.
it's just an idea.

Reply:
I did consider this variation, this reduces the stress in the code here. I wanted to avoid that, due to it is easy to overlook the fact in `_maybe_move_messages_among_queue_segments` fatal stuff can happen. In general I would incline to do as you suggested, but here I think keeping the tension is helpful fur future debugging/review as this code raises the attention of the reader.

Reply:
Thanks for the explanation. Good point about visibility, it did got my attention the first time.
I see your point, let's keep it this way then.